### PR TITLE
Fix primitives metadata

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
[jschuhmac](https://github.com/Qiskit/qiskit-terra/commits?author=jschuhmac) developed a custom estimator for his research project where additional metadata is saved, and realized that all values were updated with each iteration because the list of dicts was instantiated by reference. This PR fixes the reference `Sampler` and `Estimator` classes to avoid this problem. 

### Details and comments
Given that the only metadata currently saved are `shots`, and they don't change between circuits, it is not possible to add a unit test with this implementation, but I think this fix is good to have in case anyone wants to extend these classes and save different metadata.

### CODE REVIEW

**Comment by Cryoris** [comment]
LGTM! I would agree that this is straightforward enough to merge without a dedicated test, but let's hold merging off for a bit in case there are concerns.

**Review by t-imamichi — APPROVED** [review]
LGTM

**Review by ikkoham — APPROVED** [review]
Thank you. LGTM!

File: releasenotes/notes/fix-primitives-metadata-1e79604126e26b53.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixed a bug in the primitives' medatada where the list made copies by reference and
```
Comment (ElePT) [inline releasenotes/notes/fix-primitives-metadata-1e79604126e26b53.yaml:4]:
```suggestion
    Fixed a bug in the primitives' medatada, where the list made copies by reference and
```

File: releasenotes/notes/fix-primitives-metadata-1e79604126e26b53.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixed a bug in the primitives' medatada, where the list made copies by reference and
```
Comment (ikkoham) [inline releasenotes/notes/fix-primitives-metadata-1e79604126e26b53.yaml:4]:
```suggestion
    Fixed a bug in the primitives' metadata, where the list made copies by reference and
```

**Review by Cryoris — APPROVED** [review]

**Pull request merged**